### PR TITLE
[core] Remove unused redis code in ray python.

### PR DESCRIPTION
**Pull request by fishbone**
## Why are these changes needed?
After we decoupling redis from ray, ray doesn't need redis serve as a dependence (only redis cli in cpp) and this code is not cleaned:

- We still include redis server as a binary in pip packages
- Code starting redis server is still in ray

This PR move the code related to redis to tests and exclude redis server from the ray pip pkgs.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by jjyao — APPROVED** [review]

**Comment by iycheng** [comment]
@ericl The PR needs your approval as the code owner.

**Review by ericl — APPROVED** [review]
Stamping

**Pull request merged**
